### PR TITLE
Fixes to unittests

### PR DESCRIPTION
**Pull request by zachary-mcpher**
================ 663 passed, 8 skipped, 14 xfailed, 2 xpassed, 142 warnings in 214.74s (0:03:34) =================
(dask-ml-dev-skl024)

After installing Scikit-learn nightly build into dask-ml's ci latest env via:
`pip install --pre --extra-index https://pypi.anaconda.org/scipy-wheels-nightly/simple scikit-learn==0.24.dev0`

### CODE REVIEW

**Review by ryan-deak-zefr — APPROVED** [review]
LGTM. Are these changes backward compatible before sklearn 0.24?

**Review by zexuan-zhou — APPROVED** [review]
why do we use a dev version of sklearn instead of waiting for an official release?

File: tests/model_selection/dask_searchcv/test_model_selection.py
Comment (ryan-deak-zefr) [inline tests/model_selection/dask_searchcv/test_model_selection.py:1034]: Just checked. This should be backward compatible in sklearn 0.23.x

https://github.com/scikit-learn/scikit-learn/blob/0.23.1/sklearn/metrics/_scorer.py#L534

**Comment by zachary-mcpher** [comment]
> why do we use a dev version of sklearn instead of waiting for an official release?

Dask dask-ml maintainers use the dev build for their CI. See below:
https://github.com/dask/dask-ml/blob/v1.6.0/tests/model_selection/dask_searchcv/test_model_selection.py#L491

https://github.com/dask/dask-ml/blob/7f6dcb5b1082f5ef9f53217809efac21e0232f7f/dask_ml/_compat.py#L20

Dask-ml build logs here: https://dev.azure.com/dask-dev/dask/_build/results?buildId=1514&view=results

**Comment by zachary-mcpher** [comment]
> LGTM. Are these changes backward compatible before sklearn 0.24?

Yes they are I tested with 0.23.2 and everything passed except for the two marked xfail requiring the scikit-learn 0.24.dev0 nightly build.

**Comment by zexuan-zhou** [comment]
> https://dev.azure.com/dask-dev/dask/_build/results?buildId=1514&view=results

Got it. Thank you!

**Review by jamesbsilva — APPROVED** [review]

**Pull request merged**
